addons/sale_product_configurator/controllers/main.py

```diff
@@ -253,81 +169,193 @@ def _get_product_information(
+                so_date=so_date,
+                exclude_custom_ptav=exclude_custom_ptav,
+                **kwargs
+            ),
+            "exclusions": attribute_exclusions['exclusions'],
+            "archived_combinations": attribute_exclusions['archived_combinations'],
+            "parent_exclusions": attribute_exclusions['parent_exclusions'],
+            "quantity": quantity
+        }
+
+        return return_vals
 
-    def _get_basic_product_information(self, product_or_template, pricelist, combination, **kwargs):
+    def _get_basic_product_information(self, product_or_template, combination, pricelist=None, **kwargs):
         """ Return basic information about a product
 
         :param recordset product_or_template: The product for which to seek information, as a
                                               `product.product` or `product.template` record.
-        :param recordset|None pricelist: The pricelist to use, as a `product.pricelist` record.
         :param recordset combination: The combination of the product, as a
                                       `product.template.attribute.value` recordset.
-        :param dict kwargs: Locally unused data passed to `_get_product_price`
+        :param recordset|None pricelist: The pricelist to use, as a `product.pricelist` record.
+        :param dict kwargs: Data passed to `_get_product_price`. Contains pricelists as well
+                                    (The pricelist to use, as a `product.pricelist` record.)
         :rtype: dict
         :return: A dict with the following structure::
             {
                 'id': int,  # if product_or_template is a record of `product.product`.
                 'description_sale': str|False,
                 'display_name': str,
-                'price': float,
-                'quantity': int,
+                'price': float
             }
         """
-        basic_information = dict(
-            **product_or_template.read(['description_sale', 'display_name'])[0]
-        )
-        # If the product is a template, check the combination to compute the name to take dynamic
-        # and no_variant attributes into account. Also, drop the id which was auto-included by the
-        # search but isn't relevant since it is supposed to be the id of a `product.product` record.
-        if not product_or_template.is_product_variant:
-            basic_information['id'] = False
-            combination_name = combination._get_combination_name()
-            if combination_name:
-                basic_information.update(
-                    display_name=f"{basic_information['display_name']} ({combination_name})"
-                )
-        return dict(
-            **basic_information,
-            price=pricelist._get_product_price(
+        super_dict = super()._get_basic_product_information(product_or_template, combination, **kwargs)
+        quantity = kwargs.get('quantity')
+
+        if pricelist is None or quantity is None:
```

Comment:
We actually want `pricelist` to be `None`. It is designed to return the product price is no pricelist is selected.
```suggestion
        if pricelist is None or quantity is None:
```

Reply:
I may be wrong, but I remember experiencing errors when it was None 🤔